### PR TITLE
Added all the API on API Gateway

### PR DESCRIPTION
**Pull request by Kajal-techy**
Hello
Kindly review my PR.

### CODE REVIEW

File: src/main/java/com/apigateway/apigateway/config/SendErrorZuulCustomFilter.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        RequestContext currentContext = RequestContext.getCurrentContext();
+        HttpServletResponse response = currentContext.getResponse();
+        response.setContentType("application/json");
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        try {
+            response.getOutputStream().println(objectMapper.writeValueAsString((AuthenticationFailure.builder().message("Authentication Failure").errorCode(401)).build()));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return "null";
```
Comment (AakashIsTheLimit) [inline src/main/java/com/apigateway/apigateway/config/SendErrorZuulCustomFilter.java]: why quotes ?

File: src/main/java/com/apigateway/apigateway/config/ZuulFilterImpl.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    private String hostname;
+
+    @Value("${path.getUserByUserName}")
+    private String path;
+
+    @Value("${spring.data.authService}")
+    private String authService;
+
+    @Value("${spring.url.authServiceUserValidationPath}")
+    private String authServiceUserValidationPath;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/apigateway/apigateway/config/ZuulFilterImpl.java]: where are these used ?

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -3,9 +3,61 @@ eureka:
...
   instance:
-    instanceId: ${spring.application.name}:${random.value}
+    instanceId: ${spring.application.name}
 server:
   port: 8095
 spring:
+  data:
+    authService: auth-service
+  url:
+    authServiceUserValidationPath: /v1/validating-token
```
Comment (AakashIsTheLimit) [inline src/main/resources/application.yml]: not used anymore

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -3,9 +3,61 @@ eureka:
...
+#Routing with zuul
+zuul:
+  routes:
+    auth-token-generation:
+      path: /v1/authenticate
+      serviceId: auth-service
+      stripPrefix: false
+    auth-token-validation:
+      path: /v1/validating-token
+      sensitiveHeaders: Cookie,Set-Cookie
```
Comment (AakashIsTheLimit) [inline src/main/resources/application.yml]: curious, as to why is this header used for ?

and why we dont have that in the url above

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -3,9 +3,61 @@ eureka:
...
+      path: /v1/users/*
+      sensitiveHeaders: Cookie,Set-Cookie
+      serviceId: user-service
+      stripPrefix: false
+  SendErrorFilter:
+    error:
+      disable: true
+# Auth service
+hostname:
+  userservice: http://localhost:8080
```
Comment (AakashIsTheLimit) [inline src/main/resources/application.yml]: we should not be using any hardcoded urls

File: src/main/java/com/apigateway/apigateway/config/SendErrorZuulCustomFilter.java (hunk 1 above)
Comment (Kajal-techy) [reply src/main/java/com/apigateway/apigateway/config/SendErrorZuulCustomFilter.java:63]: Removed.

File: src/main/java/com/apigateway/apigateway/config/ZuulFilterImpl.java (hunk 2 above)
Comment (Kajal-techy) [reply src/main/java/com/apigateway/apigateway/config/ZuulFilterImpl.java:26]: Removed.

File: src/main/resources/application.yml (hunk 3 above)
Comment (Kajal-techy) [reply src/main/resources/application.yml:13]: Removed.

File: src/main/resources/application.yml (hunk 5 above)
Comment (Kajal-techy) [reply src/main/resources/application.yml:59]: Removed.

File: src/main/resources/application.yml (hunk 4 above)
Comment (Kajal-techy) [reply src/main/resources/application.yml]: Removed,  By default, headers Cookie, Set-Cookie, Authorization are blocked by the sensitiveHeaders: Cookie,Set-Cookie,Authorization configuration.

Comment (Kajal-techy) [reply src/main/resources/application.yml]: We can share headers between services in the same system, but we probably do not want sensitive headers leaking downstream into external servers

**Review by AakashIsTheLimit — APPROVED** [review]

**Pull request merged**